Lib/selectors.py

```diff
@@ -338,92 +338,90 @@ def select(self, timeout=None):
         return ready
 
 
-if hasattr(select, 'poll'):
+class _PollLikeSelector(_BaseSelectorImpl):
+    """Base class shared between poll, epoll and devpoll selectors."""
 
-    class PollSelector(_BaseSelectorImpl):
-        """Poll-based selector."""
+    def __init__(self):
+        super().__init__()
+        self._poller = self._poller()
 
-        def __init__(self):
-            super().__init__()
-            self._poll = select.poll()
+    def register(self, fileobj, events, data=None):
+        key = super().register(fileobj, events, data)
+        poller_events = 0
+        if events & EVENT_READ:
+            poller_events |= self._EVENT_READ
+        if events & EVENT_WRITE:
+            poller_events |= self._EVENT_WRITE
+        try:
+            self._poller.register(key.fd, poller_events)
+        except BaseException:
+            super().unregister(fileobj)
+            raise
+        return key
 
-        def register(self, fileobj, events, data=None):
-            key = super().register(fileobj, events, data)
-            poll_events = 0
-            if events & EVENT_READ:
-                poll_events |= select.POLLIN
-            if events & EVENT_WRITE:
-                poll_events |= select.POLLOUT
-            self._poll.register(key.fd, poll_events)
-            return key
+    def unregister(self, fileobj):
+        key = super().unregister(fileobj)
+        try:
+            self._poller.unregister(key.fd)
+        except OSError:
```

Comment:
I copied it from the previous code:
https://github.com/python/cpython/blob/2e576f5aec1f8f23f07001e2eb3db9276851a4fc/Lib/selectors.py#L423
...although the original code did that for epoll only wheres with this it's applied to all 3 selectors. Not sure if it matters. Probably not.
In general I think it would make sense to always catch OSError (also in kqueue).